File: cpp/RT_CPP/RT_CPP/directional_light.cpp
#include "directional_light.h"
#include <cfloat>

using namespace lights;


directional_light::directional_light(int id, vector3 color, vector3 direction) : light(id, color, direction)
{

}
vector3 directional_light :: get_direction (vector3 coordinate) const
{
	return direction;
}

float directional_light::light_point_distance(vector3 point)const
{
	return FLT_MAX;
}

File: Csharp/RT_CS/RT_CS/Lights/DirectionalLight.cs
using RT_CS.utility;
using System;
using System.Collections.Generic;
using System.Text;

namespace RT_CS.Lights
{
    public class DirectionalLight : Light
    {
        public DirectionalLight(int lid, Vector3 d, Vector3 c) : base(lid,d,c)
        {

        }

        public override Vector3 GetDirection(Vector3 point)
        {
            return direction;
        }

        public override float LightPointDistance(Vector3 point)
        {
            return float.PositiveInfinity;
        }
    }
}
